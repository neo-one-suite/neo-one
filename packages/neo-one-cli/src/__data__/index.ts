export * from './getClients';
export * from './getContracts';
export * from './buildTests';
export * from './generateTests';
