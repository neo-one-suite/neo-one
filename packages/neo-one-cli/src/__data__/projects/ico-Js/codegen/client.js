/* @hash 9b096a2a34b26c1ef0089640880bf614 */
// tslint:disable
/* eslint-disable */
import {
  Client,
  DapiUserAccountProvider,
  DeveloperClient,
  LocalKeyStore,
  LocalMemoryStore,
  LocalUserAccountProvider,
  NEOONEProvider,
  NEOONEDataProvider,
} from '@neo-one/client';

const getDefaultUserAccountProviders = (provider) => {
  const localUserAccountProvider = {
    memory: new LocalUserAccountProvider({
      keystore: new LocalKeyStore(new LocalMemoryStore()),
      provider,
    }),
  };

  const dapi = typeof globalThis === 'undefined' ? undefined : globalThis.neoDapi;
  if (dapi !== undefined) {
    return {
      ...localUserAccountProvider,
      dapi: new DapiUserAccountProvider({
        dapi,
        provider,
        onError: (error) => {
          throw error;
        },
      }),
    };
  }

  return localUserAccountProvider;
};

const isLocalUserAccountProvider = (userAccountProvider) => userAccountProvider instanceof LocalUserAccountProvider;

export const createClient = (getUserAccountProvidersOrHost) => {
  let getUserAccountProviders = getDefaultUserAccountProviders;
  let host = 'localhost';
  if (typeof getUserAccountProvidersOrHost === 'string') {
    host = getUserAccountProvidersOrHost;
  } else if (getUserAccountProvidersOrHost != undefined) {
    getUserAccountProviders = getUserAccountProvidersOrHost;
  }

  const providers = [];
  if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
    providers.push({ network: 'local', rpcURL: `http://${host}:10000/rpc` });
  }
  const provider = new NEOONEProvider(providers);
  const userAccountProviders = getUserAccountProviders(provider);
  const localUserAccountProviders = Object.values(userAccountProviders).filter(isLocalUserAccountProvider);
  const localUserAccountProvider = localUserAccountProviders.find(
    (userAccountProvider) => userAccountProvider.keystore instanceof LocalKeyStore,
  );
  if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
    if (localUserAccountProvider !== undefined) {
      const localKeyStore = localUserAccountProvider.keystore;
      if (localKeyStore instanceof LocalKeyStore) {
        Promise.all([
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'master',
            privateKey: 'L4qhHtwbiAMu1nrSmsTP5a3dJbxA3SNS6oheKnKd8E7KTJyCLcUv',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'alfa',
            privateKey: 'KyX5sPKRpAMb3XAFLUrHv7u1LxKkKFrpyJDgE4kceRX9FRJ4WRCQ',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'bravo',
            privateKey: 'L5LfJc2Ngsxu8ZFMvnJbYJ1QdQCstzRXmKLybsFS1aQFURkJ5CHS',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'charlie',
            privateKey: 'KxCH2Ei4TLqp2Qa7swz9bstQc5uREiCpvzvL9R6xLX8X5U8ZqeBj',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'delta',
            privateKey: 'KyVvngWhhfHiociMuwyLmGw8xTu9myKXRnvv5Fes9jDMa2Zyc6P9',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'echo',
            privateKey: 'L37qr7PWqWmgjUPfRC9mS78GjRxgGi4azySCsLUBMAa5hMka2JEm',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'foxtrot',
            privateKey: 'KwFf8gdSWxvC5Pp8AidNdF6mHqjH3CukyF3RnfwS5vzMQKLGTP13',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'golf',
            privateKey: 'Kyn2BN3QuHGYgkt9qJgvwzY8yH4xgTUAKwnGhvU1w8Nh3JnivrAr',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'hotel',
            privateKey: 'L5UXfz1xyzDkghGwistNMCV8pbpU4fg14Ez9rfo1y4KgwiadnWX3',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'india',
            privateKey: 'L5Yoq3X4ojx2FvZZxHbMcvT6var4LaXKHEpMYyyxw4jjhSUNJTRa',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'juliett',
            privateKey: 'L1DWex8PtmQJH4GYK9YAuyzmotyL6anY937LxJF54iaALrTtxsD6',
          }),
        ]).catch(() => {
          // do nothing
        });
      }
    }
  }

  return new Client(userAccountProviders);
};

export const createDeveloperClients = (host = 'localhost') => ({
  local: new DeveloperClient(new NEOONEDataProvider({ network: 'local', rpcURL: `http://${host}:10000/rpc` })),
});
