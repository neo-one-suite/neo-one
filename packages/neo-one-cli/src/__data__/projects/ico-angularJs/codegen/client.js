/* @hash 2cf036ce916c8b4860204369700ebb7b */
// tslint:disable
/* eslint-disable */
import {
  addLocalKeysSync,
  Client,
  DapiUserAccountProvider,
  DeveloperClient,
  LocalKeyStore,
  LocalMemoryStore,
  LocalUserAccountProvider,
  NEOONEProvider,
  NEOONEDataProvider,
} from '@neo-one/client';

const getDefaultUserAccountProviders = (provider) => {
  const localUserAccountProvider = {
    memory: new LocalUserAccountProvider({
      keystore: new LocalKeyStore(new LocalMemoryStore()),
      provider,
    }),
  };

  const dapi = typeof globalThis === 'undefined' ? undefined : globalThis.neoDapi;
  if (dapi !== undefined) {
    return {
      ...localUserAccountProvider,
      dapi: new DapiUserAccountProvider({
        dapi,
        provider,
        onError: (error) => {
          throw error;
        },
      }),
    };
  }

  return localUserAccountProvider;
};

const isLocalUserAccountProvider = (userAccountProvider) => userAccountProvider instanceof LocalUserAccountProvider;

export const createClient = (getUserAccountProvidersOrHost) => {
  let getUserAccountProviders = getDefaultUserAccountProviders;
  let host = 'localhost';
  if (typeof getUserAccountProvidersOrHost === 'string') {
    host = getUserAccountProvidersOrHost;
  } else if (getUserAccountProvidersOrHost != undefined) {
    getUserAccountProviders = getUserAccountProvidersOrHost;
  }

  const providers = [];
  if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
    providers.push({ network: 'local', rpcURL: `http://${host}:10040/rpc` });
  }
  const provider = new NEOONEProvider(providers);
  const userAccountProviders = getUserAccountProviders(provider);
  const localUserAccountProviders = Object.values(userAccountProviders).filter(isLocalUserAccountProvider);
  const localUserAccountProvider = localUserAccountProviders.find(
    (userAccountProvider) => userAccountProvider.keystore instanceof LocalKeyStore,
  );
  if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
    if (localUserAccountProvider !== undefined) {
      const localKeyStore = localUserAccountProvider.keystore;
      if (localKeyStore instanceof LocalKeyStore) {
        addLocalKeysSync(
          [
            { network: 'local', name: 'master', privateKey: 'L4qhHtwbiAMu1nrSmsTP5a3dJbxA3SNS6oheKnKd8E7KTJyCLcUv' },
            { network: 'local', name: 'alfa', privateKey: 'KyX5sPKRpAMb3XAFLUrHv7u1LxKkKFrpyJDgE4kceRX9FRJ4WRCQ' },
            { network: 'local', name: 'bravo', privateKey: 'L5LfJc2Ngsxu8ZFMvnJbYJ1QdQCstzRXmKLybsFS1aQFURkJ5CHS' },
            { network: 'local', name: 'charlie', privateKey: 'KxCH2Ei4TLqp2Qa7swz9bstQc5uREiCpvzvL9R6xLX8X5U8ZqeBj' },
            { network: 'local', name: 'delta', privateKey: 'KyVvngWhhfHiociMuwyLmGw8xTu9myKXRnvv5Fes9jDMa2Zyc6P9' },
            { network: 'local', name: 'echo', privateKey: 'L37qr7PWqWmgjUPfRC9mS78GjRxgGi4azySCsLUBMAa5hMka2JEm' },
            { network: 'local', name: 'foxtrot', privateKey: 'KwFf8gdSWxvC5Pp8AidNdF6mHqjH3CukyF3RnfwS5vzMQKLGTP13' },
            { network: 'local', name: 'golf', privateKey: 'Kyn2BN3QuHGYgkt9qJgvwzY8yH4xgTUAKwnGhvU1w8Nh3JnivrAr' },
            { network: 'local', name: 'hotel', privateKey: 'L5UXfz1xyzDkghGwistNMCV8pbpU4fg14Ez9rfo1y4KgwiadnWX3' },
            { network: 'local', name: 'india', privateKey: 'L5Yoq3X4ojx2FvZZxHbMcvT6var4LaXKHEpMYyyxw4jjhSUNJTRa' },
            { network: 'local', name: 'juliett', privateKey: 'L1DWex8PtmQJH4GYK9YAuyzmotyL6anY937LxJF54iaALrTtxsD6' },
          ],
          localKeyStore,
        );
      }
    }
  }

  return new Client(userAccountProviders);
};

export const createDeveloperClients = (host = 'localhost') => ({
  local: new DeveloperClient(new NEOONEDataProvider({ network: 'local', rpcURL: `http://${host}:10040/rpc` })),
});
