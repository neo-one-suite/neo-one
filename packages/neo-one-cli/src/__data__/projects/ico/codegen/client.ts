/* @hash 3dd29714fc27a507c2e0a32a2cb1e50e */
// tslint:disable
/* eslint-disable */
import {
  Client,
  DapiUserAccountProvider,
  DeveloperClient,
  DeveloperClients,
  LocalKeyStore,
  LocalMemoryStore,
  LocalUserAccountProvider,
  NEOONEProvider,
  NEOONEDataProvider,
  UserAccountProviders,
} from '@neo-one/client';

export interface DefaultUserAccountProviders {
  readonly memory: LocalUserAccountProvider<LocalKeyStore, NEOONEProvider>;
}

const getDefaultUserAccountProviders = (provider: NEOONEProvider) => {
  const localUserAccountProvider = {
    memory: new LocalUserAccountProvider({
      keystore: new LocalKeyStore(new LocalMemoryStore()),
      provider,
    }),
  };

  const dapi = typeof globalThis === 'undefined' ? undefined : (globalThis as any).neoDapi;
  if (dapi !== undefined) {
    return {
      ...localUserAccountProvider,
      dapi: new DapiUserAccountProvider({
        dapi,
        provider,
        onError: (error) => {
          throw error;
        },
      }),
    };
  }

  return localUserAccountProvider;
};

const isLocalUserAccountProvider = (userAccountProvider: any): userAccountProvider is LocalUserAccountProvider =>
  userAccountProvider instanceof LocalUserAccountProvider;

export const createClient = <TUserAccountProviders extends UserAccountProviders<any> = DefaultUserAccountProviders>(
  getUserAccountProvidersOrHost:
    | string
    | ((provider: NEOONEProvider) => TUserAccountProviders) = getDefaultUserAccountProviders as any,
): Client<
  TUserAccountProviders extends UserAccountProviders<infer TUserAccountProvider> ? TUserAccountProvider : any,
  TUserAccountProviders
> => {
  let getUserAccountProviders = getDefaultUserAccountProviders;
  let host = 'localhost';
  if (typeof getUserAccountProvidersOrHost === 'string') {
    host = getUserAccountProvidersOrHost;
  } else if (getUserAccountProvidersOrHost != undefined) {
    getUserAccountProviders = getUserAccountProvidersOrHost as any;
  }

  const providers = [];
  if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
    providers.push({ network: 'local', rpcURL: `http://${host}:10010/rpc` });
  }
  const provider = new NEOONEProvider(providers);
  const userAccountProviders = getUserAccountProviders(provider);
  const localUserAccountProviders = Object.values(userAccountProviders).filter(
    isLocalUserAccountProvider,
  ) as LocalUserAccountProvider[];
  const localUserAccountProvider = localUserAccountProviders.find(
    (userAccountProvider) => userAccountProvider.keystore instanceof LocalKeyStore,
  );
  if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
    if (localUserAccountProvider !== undefined) {
      const localKeyStore = localUserAccountProvider.keystore;
      if (localKeyStore instanceof LocalKeyStore) {
        Promise.all([
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'master',
            privateKey: 'L4qhHtwbiAMu1nrSmsTP5a3dJbxA3SNS6oheKnKd8E7KTJyCLcUv',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'alfa',
            privateKey: 'KyX5sPKRpAMb3XAFLUrHv7u1LxKkKFrpyJDgE4kceRX9FRJ4WRCQ',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'bravo',
            privateKey: 'L5LfJc2Ngsxu8ZFMvnJbYJ1QdQCstzRXmKLybsFS1aQFURkJ5CHS',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'charlie',
            privateKey: 'KxCH2Ei4TLqp2Qa7swz9bstQc5uREiCpvzvL9R6xLX8X5U8ZqeBj',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'delta',
            privateKey: 'KyVvngWhhfHiociMuwyLmGw8xTu9myKXRnvv5Fes9jDMa2Zyc6P9',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'echo',
            privateKey: 'L37qr7PWqWmgjUPfRC9mS78GjRxgGi4azySCsLUBMAa5hMka2JEm',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'foxtrot',
            privateKey: 'KwFf8gdSWxvC5Pp8AidNdF6mHqjH3CukyF3RnfwS5vzMQKLGTP13',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'golf',
            privateKey: 'Kyn2BN3QuHGYgkt9qJgvwzY8yH4xgTUAKwnGhvU1w8Nh3JnivrAr',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'hotel',
            privateKey: 'L5UXfz1xyzDkghGwistNMCV8pbpU4fg14Ez9rfo1y4KgwiadnWX3',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'india',
            privateKey: 'L5Yoq3X4ojx2FvZZxHbMcvT6var4LaXKHEpMYyyxw4jjhSUNJTRa',
          }),
          localKeyStore.addUserAccount({
            network: 'local',
            name: 'juliett',
            privateKey: 'L1DWex8PtmQJH4GYK9YAuyzmotyL6anY937LxJF54iaALrTtxsD6',
          }),
        ]).catch(() => {
          // do nothing
        });
      }
    }
  }

  return new Client(userAccountProviders as any);
};

export const createDeveloperClients = (host = 'localhost'): DeveloperClients => ({
  local: new DeveloperClient(new NEOONEDataProvider({ network: 'local', rpcURL: `http://${host}:10010/rpc` })),
});
