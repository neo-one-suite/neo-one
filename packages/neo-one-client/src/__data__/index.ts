export * from './keys';
export * from './transactions';
