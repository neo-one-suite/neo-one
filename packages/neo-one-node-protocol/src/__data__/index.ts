export { context } from './context';
export { keys } from './keys';
