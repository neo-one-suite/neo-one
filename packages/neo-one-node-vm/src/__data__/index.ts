import * as assets from './assets';
import * as transactions from './transactions';
import * as testUtils from './utils';

export * from './createBlockchain';
export * from './keys';

export { assets, testUtils, transactions };
