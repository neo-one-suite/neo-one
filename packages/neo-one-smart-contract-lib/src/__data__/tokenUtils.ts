import BigNumber from 'bignumber.js';
import {
  addressToScriptHash,
  createPrivateKey,
  privateKeyToScriptHash,
  TransactionResult,
  InvokeReceipt,
  UserAccountID,
  SmartContract,
} from '@neo-one/client';
import { cleanupTest, setupContractTest } from '@neo-one/smart-contract-test';

import appRootDir from 'app-root-dir';
import path from 'path';

export interface DeployOptions {
  masterPrivateKey: string;
  masterAccountID: UserAccountID;
  smartContract: SmartContract;
}

export interface Options {
  contractName: string;
  name: string;
  symbol: string;
  decimals: number;
  deploy: (options: DeployOptions) => Promise<TransactionResult<InvokeReceipt>>;
  issueValue: BigNumber;
  transferValue: BigNumber;
  contractHash: string;
  dir: string;
}

export const testToken = ({
  contractName,
  name,
  symbol,
  decimals,
  deploy,
  issueValue,
  transferValue,
  contractHash,
  dir,
}: Options) => {
  const setup = () =>
    setupContractTest({
      dir: path.resolve(
        appRootDir.get(),
        'packages',
        'neo-one-smart-contract-lib',
        'src',
        '__data__',
        'contracts',
      ),
      contractName,
    });

  describe(contractName, () => {
    afterEach(async () => {
      await cleanupTest();
    });

    test('properties + issue + balanceOf + totalSupply + transfer', async () => {
      const {
        networkName,
        keystore,
        developerClient,
        smartContract,
        masterAccountID,
        masterPrivateKey,
      } = await setup();

      const [nameResult, decimalsResult, symbolResult] = await Promise.all([
        smartContract.name(),
        smartContract.decimals(),
        smartContract.symbol(),
      ]);
      expect(nameResult).toEqual(name);
      expect(decimalsResult.toString()).toEqual(`${decimals}`);
      expect(symbolResult).toEqual(symbol);

      const wallet0 = await keystore.addAccount({
        network: networkName,
        name: 'wallet0',
        privateKey: createPrivateKey(),
      });
      const account0 = wallet0.account.id;

      let result = await deploy({
        masterPrivateKey,
        masterAccountID,
        smartContract,
      });
      let [receipt] = await Promise.all([
        result.confirmed({ timeoutMS: 30000 }),
        developerClient.runConsensusNow(),
      ]);

      if (receipt.result.state !== 'HALT') {
        expect(receipt.result.state).toEqual('HALT');
        throw new Error('For TS');
      }

      expect(receipt.result.gasConsumed.toString()).toMatchSnapshot(
        'deploy consumed',
      );
      expect(receipt.result.gasCost.toString()).toMatchSnapshot('deploy cost');
      expect(receipt.result.value).toBeTruthy();
      expect(receipt.events).toHaveLength(1);
      let event = receipt.events[0];
      expect(event.name).toEqual('transfer');
      expect(event.parameters.from).toEqual(contractHash);
      expect(event.parameters.to).toEqual(
        privateKeyToScriptHash(masterPrivateKey),
      );
      if (event.parameters.amount == null) {
        expect(event.parameters.amount).toBeTruthy();
        throw new Error('For TS');
      }
      expect(event.parameters.amount.toString()).toEqual(issueValue.toString());

      let value = await smartContract.balanceOf(
        addressToScriptHash(masterAccountID.address),
      );
      expect(value.toString()).toEqual(issueValue.toString());

      value = await smartContract.totalSupply();
      expect(value.toString()).toEqual(issueValue.toString());

      result = await smartContract.transfer(
        addressToScriptHash(masterAccountID.address),
        addressToScriptHash(account0.address),
        transferValue,
        { from: masterAccountID },
      );
      [receipt] = await Promise.all([
        result.confirmed(),
        developerClient.runConsensusNow(),
      ]);

      if (receipt.result.state !== 'HALT') {
        expect(receipt.result.state).toEqual('HALT');
        throw new Error('For TS');
      }

      expect(receipt.result.gasConsumed.toString()).toMatchSnapshot(
        'transfer consume',
      );
      expect(receipt.result.gasCost.toString()).toMatchSnapshot(
        'transfer cost',
      );
      expect(receipt.events).toHaveLength(1);
      event = receipt.events[0];
      expect(event.name).toEqual('transfer');
      expect(event.parameters.from).toEqual(
        privateKeyToScriptHash(masterPrivateKey),
      );
      expect(event.parameters.to).toEqual(
        addressToScriptHash(account0.address),
      );
      if (event.parameters.amount == null) {
        expect(event.parameters.amount).toBeTruthy();
        throw new Error('For TS');
      }
      expect(event.parameters.amount.toString()).toEqual(
        transferValue.toString(),
      );

      const remainingValue = issueValue.minus(transferValue);
      value = await smartContract.balanceOf(
        addressToScriptHash(masterAccountID.address),
      );
      expect(value.toString()).toEqual(remainingValue.toString());

      value = await smartContract.balanceOf(
        addressToScriptHash(account0.address),
      );
      expect(value.toString()).toEqual(transferValue.toString());

      value = await smartContract.totalSupply();
      expect(value.toString()).toEqual(issueValue.toString());

      value = await smartContract.balanceOf(
        privateKeyToScriptHash(createPrivateKey()),
      );
      expect(value.toString()).toEqual('0');
    });
  });
};
