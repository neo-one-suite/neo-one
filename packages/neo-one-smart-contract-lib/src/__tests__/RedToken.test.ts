import BigNumber from 'bignumber.js';
import { privateKeyToScriptHash } from '@neo-one/client';

import { testToken } from '../__data__';

const issueValue = new BigNumber('1000000');

testToken({
  contractName: 'RedToken',
  name: 'RedToken',
  symbol: 'RT',
  decimals: 8,
  deploy: async ({ masterPrivateKey, masterAccountID, smartContract }) =>
    smartContract.deploy(privateKeyToScriptHash(masterPrivateKey), issueValue, {
      from: masterAccountID,
    }),
  issueValue,
  transferValue: new BigNumber('10'),
  contractHash: '0xa5e724ae2a146f9a9ac9bc44f9d82ad9bf14ccd1',
  dir: 'red',
});
