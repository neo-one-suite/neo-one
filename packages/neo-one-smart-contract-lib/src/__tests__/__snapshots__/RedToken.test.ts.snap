// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedToken properties + issue + balanceOf + totalSupply + transfer: deploy consumed 1`] = `"11.233"`;

exports[`RedToken properties + issue + balanceOf + totalSupply + transfer: deploy cost 1`] = `"21.233"`;

exports[`RedToken properties + issue + balanceOf + totalSupply + transfer: transfer consume 1`] = `"8.466"`;

exports[`RedToken properties + issue + balanceOf + totalSupply + transfer: transfer cost 1`] = `"18.466"`;
